libraries/chain/include/eosio/chain/code_object.hpp

```diff
@@ -33,8 +35,23 @@ namespace eosio { namespace chain {
       >
    >;
 
+   struct snapshot_code_object_v1 {
+      static constexpr uint32_t minimum_version = 0;
+      static constexpr uint32_t maximum_version = 8;
+      static_assert(chain_snapshot_header::minimum_compatible_version <= maximum_version,
+                    "snapshot_code_object_v1 is no longer needed");
+
+      digest_type  code_hash;
+      shared_blob  code;
+      uint64_t     code_ref_count;
+      uint32_t     first_block_used;
+      uint8_t      vm_type = 0;
+      uint8_t      vm_version = 0;
+   };
+
```

Comment:
I'm a little confused on this new `snapshot_code_object_v1` because it's the same as the `code_object` previously was serialized. Also, when we're converting from different chainbase<->snapshot types usually (always?) we use `snapshot_row_traits` which I don't see anywhere in here.

I'm suspicious we don't need any of these snapshot changes and can instead add a `code_object::reflector_init()` which just populates the new `sync_call_supported` field. (this would mean that we'd not include `sync_call_supported` as part of the `fc_reflect` which is a little weird but not unheard of)

I'm not 100% confident on the above but I'd suggest trying it

Reply:
The new `snapshot_code_object_v1` is for snapshot versions prior to sync call protocol feature. We have similar structs for block_state and chain_config.

Will try using `code_object::reflector_init()`. Thanks.

Reply:
Thanks!

`code_object::reflector_init()` works. We don't need to place `sync_call_supported` into snapshots.